### PR TITLE
AVM: Provide access to some more block header values

### PR DESCRIPTION
**Pull request by jannotti**
Because of block incentives, there are good reasons to expose the FeeSink and the ProposerPayout from the block header to AVM using the block opcode.  While doing that, we can consider a few more things that might be useful.

Currently this PR exposes these.  
```
// BlkBranch is the hash of the previous block
BlkBranch

// BlkFeeSink is the fee sink for the given round
BlkFeeSink

// BlkProtocol is the ConsensusVersion of the block.
BlkProtocol

// BlkTxnCounter is the number of the next transaction after the block
BlkTxnCounter

// BlkProposerPayout is the actual amount moved from feesink to proposer
BlkProposerPayout
```

### CODE REVIEW

**Review by algorandskiy — DISMISSED** [review]

File: data/transactions/logic/fields.go [hunk 1]
```diff
@@ -991,12 +991,24 @@ const (
...
 	BlkTimestamp
+
+	// BlkBranch is the hash of the previous block
+	BlkBranch
+	// BlkFeeSink is the fee sink for the given round
+	BlkFeeSink
+	// BlkProtocol is the ConsensusVersion of the block.
+	BlkProtocol
+	// BlkTxnCounter is the number of the next transaction after the block
+	BlkTxnCounter
```
[the remark this answers is not in the document]
Comment (jannotti) [reply data/transactions/logic/fields.go:1002]: That's fair, I suppose there is more extensive test usage out there for this. I'll rearrange,

File: data/transactions/logic/fields.go [hunk 2]
```diff
@@ -1010,11 +1022,18 @@ type blockFieldSpec struct {
...
 
 var blockFieldSpecs = [...]blockFieldSpec{
-	{BlkSeed, StackBytes, randomnessVersion},
+	{BlkSeed, StackBytes32, randomnessVersion},
 	{BlkTimestamp, StackUint64, randomnessVersion},
+
+	{BlkBranch, StackBytes32, 11},
+	{BlkFeeSink, StackAddress, 11},
+	{BlkProtocol, StackBytes, 11},
+	{BlkTxnCounter, StackUint64, 11},
```
[the remark this answers is not in the document]
Comment (jannotti) [reply data/transactions/logic/fields.go]: done

**Review by jasonpaulos — APPROVED** [review]

File: test/scripts/e2e_subs/hdr-access.py [hunk 3]
```diff
@@ -85,11 +81,33 @@
...
+"""
+txinfo, err = goal.app_create(joe, goal.assemble(teal), lifetime=100)
+assert not err, err
+assert len(txinfo["logs"]) == 1
+opcode = txinfo["logs"][0]
+
+block = goal.algod.block_info(txinfo['confirmed-round']-2)['block']
+api = base64.b64encode(enc.decode_address(block['fees'])).decode("utf-8")
+
+print(opcode, api)
```
Comment (algorandskiy) [inline test/scripts/e2e_subs/hdr-access.py:108]: forgotten debug print?

**Review by algorandskiy — APPROVED** [review]

File: test/scripts/e2e_subs/hdr-access.py (hunk 3 above)
Comment (jannotti) [reply test/scripts/e2e_subs/hdr-access.py:108]: Yes and no. Python's assert does not print the objects being compared, so this is helpful in the case of failure. When the test succeeds, the output is surpressed anyway.

I'll remove it if you'd prefer it gone, let me know.

**Review by onetechnical — APPROVED** [review]

**Pull request merged**